src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -71,6 +71,7 @@ struct Parameters
 
   Real sparsity;
 
+  // Maybe shared pointer?
   Topology_t* potentialPool;
```

Comment:
even `unique_ptr`, as only this instance uses potentialPool topology.

Reply:
I decided to change all of the topology subclasses to be pure functions.  IMO in C++ the syntax for lambda's is a little uglier than for classes, but I think that conceptually functions are far better suited for this task.